packages/fxa-auth-server/lib/routes/emails.js

```diff
@@ -712,59 +668,43 @@ module.exports = (
         const uid = sessionToken.uid;
         const primaryEmail = sessionToken.email;
         const email = request.payload.email;
-        let account;
-
-        return customs
-          .check(request, primaryEmail, 'deleteEmail')
-          .then(() => {
-            return db.account(uid);
-          })
-          .then(result => {
-            account = result;
-
-            if (sessionToken.tokenVerificationId) {
-              throw error.unverifiedSession();
-            }
-          })
-          .then(deleteEmail)
-          .then(resetAccountTokens)
-          .then(() => {
-            // Find the email object that corresponds to the email being deleted
-            const emailIsVerified = account.emails.find(item => {
-              return (
-                item.normalizedEmail === email.toLowerCase() && item.isVerified
-              );
-            });
 
-            // Don't bother sending a notification if removing an email that was never verified
-            if (!emailIsVerified) {
-              return P.resolve();
-            }
-
-            // Notify only primary email and all *other* verified secondary emails about the
-            // deletion.
-            const emails = account.emails.filter(item => {
-              if (item.normalizedEmail !== email.toLowerCase()) {
-                return item;
-              }
-            });
-            return mailer.sendPostRemoveSecondaryEmail(emails, account, {
-              deviceId: sessionToken.deviceId,
-              secondaryEmail: email,
-              uid,
-            });
-          })
-          .then(() => {
-            return {};
-          });
+        await customs.check(request, primaryEmail, 'deleteEmail');
+        const account = await db.account(uid);
 
-        function deleteEmail() {
-          return db.deleteEmail(uid, email.toLowerCase());
+        if (sessionToken.tokenVerificationId) {
+          throw error.unverifiedSession();
         }
 
-        function resetAccountTokens() {
-          return db.resetAccountTokens(uid);
+        await db.deleteEmail(uid, email.toLowerCase());
+        await db.resetAccountTokens(uid);
+
+        // Find the email object that corresponds to the email being deleted
+        const emailIsVerified = account.emails.find(item => {
+          return (
+            item.normalizedEmail === email.toLowerCase() && item.isVerified
+          );
+        });
+
+        // Don't bother sending a notification if removing an email that was never verified
+        if (!emailIsVerified) {
+          return {};
         }
+
+        // Notify only primary email and all *other* verified secondary emails about the
```

Comment:
This comment is a bit confusing, would it be more clear as `Notify any verified email address associated with the account of the deletion`